extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -27,12 +35,42 @@ public fun MapboxMap(
    */
   modifier: Modifier = Modifier,
   /**
-   * Defines the initialisation configurations for a [MapboxMap].
+   * Defines the initialisation configurations factory for a [MapboxMap].
    *
    * It can only be set once and not mutable after the initialisation. Mutating the [MapInitOptions]
    * during recomposition will result in a [IllegalStateException].
    */
-  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current.applicationContext),
+  mapInitOptionsFactory: (Context) -> MapInitOptions = { context -> MapInitOptions(context) },
+  /**
+   * Settings for showing the attribution icon on the map.
+   */
+  attributionSettings: AttributionSettings = AttributionSettings(),
+  /**
+   * Settings for showing the compass icon on the map.
+   */
+  compassSettings: CompassSettings = CompassSettings(),
+  /**
+   * Gesture configuration allows to control the user touch interaction.
+   */
+  gesturesSettings: GesturesSettings = GesturesSettings(),
+  /**
+   * Settings for showing a location puck on the map.
+   */
+  locationComponentSettings: LocationComponentSettings = LocationComponentSettingsProvider.getDefaultSettings(
+    context = LocalContext.current.applicationContext,
+  ),
+  /**
+   * Additional settings for showing a location puck on the map.
+   */
+  locationComponentSettings2: LocationComponentSettings2 = LocationComponentSettings2(),
```

Comment:
Is LocationComponentSettings2 needed here, couldn't all location component settings be part of LocationComponentSettings since there's not backward compatibility issues?

Reply:
No, `LocationComponentSettings` and `LocationComponentSettings2` are data classes, and they don't overlap, these should be fixed in v11.

Reply:
Yeah I've thought it's a copy of the original data classes used here, but it's not, I see now